### ISSUE TITLE
form.reset() generates React warning (An update (setState, replaceState, or forceUpdate) was scheduled from inside an update function.)

### ISSUE THREAD

**Issue by philipmjohnson** [vazco/uniforms#395]
I created a React.Component with a form, and want to clear the form after a successful submit.  To do this, I create a form ref. However, when the form resets, React generates the following warning:

```
Warning: An update (setState, replaceState, or forceUpdate) was scheduled from inside 
an update function. Update functions should be pure, with zero side-effects. 
Consider using componentDidUpdate or a callback.
```

The Component code that generates this warning is [AddStuff.jsx](https://github.com/ics-software-engineering/meteor-application-template-react/blob/master/app/imports/ui/pages/AddStuff.jsx)

You can reproduce the error by installing and running meteor-application-template-react. The installation instructions are [here](https://ics-software-engineering.github.io/meteor-application-template-react/).

**Comment by andrewckor** [comment]
I have the same issue for a long time but I thought that's my fault.

**Comment by radekmie** [comment]
Thanks @philipmjohnson for reporting that (and using Gitter too!). I've digged in a bit yesterday and found out, that in some newer version on React calling `setState` from `setState` function triggers such warning. Bug [source](https://github.com/vazco/uniforms/blob/master/packages/uniforms/src/AutoForm.js#L77).

I'm planing to post a patch and do a pre-release today or tomorrow.

---

By the way - nice work with that template!

**Comment by radekmie** [comment]
Should be fixed in `v1.23.1`.

**Comment by radekmie** [comment]
I assume it's working.
